api/query_engine.go

```diff
@@ -111,6 +125,30 @@ func alignRequests(reqs []models.Req, aggSettings mdata.AggSettings) ([]models.R
 
 	chosenInterval := options[selected].interval
 
+	// by default we select the lowest res (longest time range) option.
+	selectedHighestRes := len(options) - 1
+	// then we loop over the remaining options trying to find the highest res where the
+	// oldest timestamp is still older than our From ts
+	for i := len(options) - 2; i >= 0; i-- {
+		if oldestTs(options[i].ttl) > reqs[0].From {
+			break
+		}
+		selectedHighestRes = i
+	}
+
+	// record how many datapoints will be sent to graphite-api if we use the old-style selection mechanism
+	dpAdjGranularity.SetUint32(tsRange / chosenInterval)
+
+	// record the aggregation index that has been selected
+	aggAdjGranularity.SetUint32(uint32(selected))
```

Comment:
there's another reason why they are wrong. it reports the data being fetched, not what we return to graphite (which is subject to runtime consolidation). I'll fix that too. I think it makes sense to report the poincounts both for what MT fetches as well as what it delivers to graphite (which is the same in the MaxRes case, but not in the current case)